### DISCUSSION TITLE
How to rotate the scene when I update the 3D scene by plt.render()?

### DISCUSSION THREAD

**Discussion by CCHLENG** [marcomusy/vedo#755] [Q&A] [answered]
Hi!
Now I'm working on a kind of Lidar, I need to display the points cloud in real time(3600*32 points per frame, 20fps max). Meanwhile I need to draw some box(about 30 boxes per frame, number of box is different in each frames) like following figure.
To update points, I can use `pointcloud.Points()`. To draw boxes in real time, which api should I use?

![image](https://user-images.githubusercontent.com/25116183/207259171-9befd9cb-bd5d-459b-bbbd-72d09ae1fe1c.png)

I want to rotate or pan the scene when I update the 3D scene by following codes:

```python
...
pts = pointcloud.Points(pt)
plt = show( pts, axes=5, interactive=False)
for i in range(1000):
    pt = np.random.uniform(10, 100, size=(1000, 3))
    pts.points(pt)
    plt.render()
plt.interactive().close()
```
But it not works, I can't rotate or pan the scene when 3D scene is updating. 
It may caused by `interactive=False`. I need threading to do that? Is there an easy way or some demos I can reference?
Thanks!

**Comment by marcomusy** [comment, marked as answer]
This is only possible with vtk<9.2.2 due to a bug in vtk. Try

`pip install vtk==9.0.3`

then

```python
from vedo import *

settings.allow_interaction = True

pts = np.random.uniform(10, 100, size=(100000, 3))
vpts = Points(pts, c='white', r=1)

plt = Plotter(axes=5, interactive=False, bg='black')
plt.show(vpts)

pb = ProgressBar(0,500, c='g')
for i in range(500):
    pts += np.random.randn(100000, 3)/20
    vpts.points(pts)

    cube_centers = np.random.uniform(10, 100, size=(10, 3))
    boxes=[]
    for cc in cube_centers:
        box = Box(size=(30,20,10), c='green5')
        ax, ay = np.random.uniform(0, 45, size=2)
        box.rotate_x(ax).rotate_y(ay).pos(cc)
        box.wireframe().lw(2).lighting("off")
        boxes.append(box)
    plt.remove("Box").add(boxes)
    # plt.render() # no need! add already calls it
    pb.print()
    if plt.escaped: # becomes True if Esc is pressed
        break

plt.interactive().close()
```
![Screenshot from 2022-12-13 19-19-53](https://user-images.githubusercontent.com/32848391/207414184-0765031e-b2f9-4c5e-934e-6e344af2e350.png)

on my machine I get 56 fps.
Note that when moving the scene the run flow is held

**Reply by CCHLENG** [reply]
Thanks for your reply!  your example works as expected on my computer.
On my computer (i7 8700k + GTX1080Ti; vedo == 2022.4.1, vtk == 9.0.3), I get 32fps. 
How can I run it faster?

```python
━━━━━━━━━━━━━━━━━━━━━━ elapsed: 16s (32.1 it/s)        
```

Is there a way to moving the scene during the run flow? Like pyqtgraph or pyvistaqt.
Thanks!

**Comment by marcomusy** [comment]
> How can I run it faster?

are you sure you removed the call to `render()` ?

try also
`pip install -U git+https://github.com/marcomusy/vedo.git`

I don't see how can be made faster..

> Is there a way to moving the scene during the run flow? Like pyqtgraph or pyvistaqt.

No, sorry! At the minute `allow_interaction=True` is the only option.

**Reply by CCHLENG** [reply]
OK, thanks!
